transaction/core/src/validation/error.rs

```diff
@@ -1,134 +1,104 @@
 // Copyright (c) 2018-2020 MobileCoin Inc.
 
 use alloc::string::String;
-use failure::Fail;
+use displaydoc::Display;
 use mc_crypto_keys::KeyError;
 use serde::{Deserialize, Serialize};
 
 /// Type alias for transaction validation results.
 pub type TransactionValidationResult<T> = Result<T, TransactionValidationError>;
 
 /// Reasons why a single transaction may fail to be valid with respect to the current ledger.
-#[derive(Clone, Debug, Eq, Fail, Hash, Ord, PartialEq, PartialOrd, Serialize, Deserialize)]
+#[derive(Clone, Debug, Display, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize, Deserialize)]
 pub enum TransactionValidationError {
     /// Each input should have one membership proof.
-    #[fail(display = "InputsProofsLengthMismatch")]
     InputsProofsLengthMismatch,
 
     /// A transaction must have at least one input.
-    #[fail(display = "NoInputs")]
     NoInputs,
 
     /// A transaction must have no more than the maximum allowed number of inputs.
-    #[fail(display = "TooManyInputs")]
     TooManyInputs,
 
     /// Each input must have a signature.
-    #[fail(display = "InsufficientInputSignatures")]
     InsufficientInputSignatures,
 
     /// Each input must have a valid signature.
-    #[fail(display = "InvalidInputSignature")]
     InvalidInputSignature,
 
     /// The transaction must have a valid RingCT signature.
-    #[fail(display = "InvalidTransactionSignature")]
-    InvalidTransactionSignature(#[fail(cause)] crate::ring_signature::Error),
+    InvalidTransactionSignature(crate::ring_signature::Error),
```

Comment:
The text comment should include the specific error (e.g. `The transaction must have a valid RingCT signature: {0}`